polaris-core/src/main/java/org/apache/polaris/core/entity/CatalogEntity.java

```diff
@@ -240,6 +246,21 @@ public Builder setStorageConfigurationInfo(
         }
         allowedLocations.add(defaultBaseLocation);
         validateMaxAllowedLocations(allowedLocations);
+
+        PolarisStorageConfigurationInfo existingConfig = null;
```

Comment:
@flyrain would the DAO be a better candidate place for this logic?

Reply:
Agreed. It'd be nice the logic was wrapped in a DAO method like `CatalogDao::GetExistingStorageConfig()`

Reply:
The logic is currently in the CatalogEntity builder, for the builder, it doesn't have the catalog entity but just an internalProperties map. I'm trying to get the storage config from the internal properties map. Do you mean I need to implement a method like this:

```java
class CatalogDao {
  PolarisStorageConfigurationInfo getExistingStorageConfig(Map<String, String> internalProperties) {
  }
}
```